unified-energy/src/main/java/io/github/unifiedtechpower/unified/energy/manager/UnifiedEnergy.java

```diff
@@ -0,0 +1,109 @@
+package io.github.unifiedtechpower.unified.energy.manager;
+
+import io.github.unifiedtechpower.unified.energy.storage.EnergyStorage;
+import org.bukkit.Chunk;
+import org.bukkit.Location;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Manager for different {@link EnergyNetworkManager} implementations.
+ */
+public class UnifiedEnergy {
+    
+    private static final UnifiedEnergy INSTANCE = new UnifiedEnergy();
+    
+    private final HashSet<EnergyNetworkManager> managers = new HashSet<>();
+    
+    private UnifiedEnergy() {
+    }
+    
+    /**
+     * Adds a {@link EnergyNetworkManager} to the set of managers.
+     *
+     * @param manager the {@link EnergyNetworkManager} to add
+     * @return true if the manager was added, false if it was already in the set
+     */
+    public boolean addManager(@NotNull EnergyNetworkManager manager) {
```

Comment:
Do we want some kind of removeManager method?

I can't think of many uses for it, but I also think it might be better to have it encase someone finds a use case, like disabling a plugin.

Reply:
Yeah that might be needed. I'll add it